dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/put-fee-record-correction-transient-form-data.controller/helpers/validate-record-correction-transient-form-values.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  isCurrencyValid,
+  isMonetaryAmountValid,
+  RECORD_CORRECTION_REASON,
+  RecordCorrectionFormValues,
+  RecordCorrectionFormValueValidationErrors,
+  RecordCorrectionReason,
+} from '@ukef/dtfs2-common';
+import { difference } from 'lodash';
+import { getAdditionalCommentsValidationError, getFacilityIdValidationError } from './get-record-correction-transient-form-validation-error';
+
+/**
+ * Gets the form value associated with a specific record correction reason.
+ * @param formValues - The form values containing the correction data.
+ * @param reason - The record correction reason to get the form value for.
+ * @returns The form value for the specified reason, or undefined if no value exists.
+ * @throws Error if an invalid record correction reason is provided.
+ */
+export const getFormValueForReason = (formValues: RecordCorrectionFormValues, reason: RecordCorrectionReason): string | undefined => {
+  switch (reason) {
+    case RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT:
+      return formValues.facilityId;
+    case RECORD_CORRECTION_REASON.REPORTED_CURRENCY_INCORRECT:
+      return formValues.reportedCurrency;
+    case RECORD_CORRECTION_REASON.REPORTED_FEE_INCORRECT:
+      return formValues.reportedFee;
+    case RECORD_CORRECTION_REASON.UTILISATION_INCORRECT:
+      return formValues.utilisation;
+    case RECORD_CORRECTION_REASON.OTHER:
+      return formValues.additionalComments;
+    default:
+      throw new Error(`Invalid record correction reason: ${reason}`);
+  }
+};
+
+/**
+ * Validates that there are no form values provided for reasons that are not in the correction request.
+ *
+ * Note that the "additionalComments" field is required when {@link RECORD_CORRECTION_REASON.OTHER} is
+ * one of the reasons, and optional when {@link RECORD_CORRECTION_REASON.OTHER} is not one of the
+ * reasons. It is therefore not included in the "unexpected reasons" check as it can be provided
+ * regardless of the correction reasons.
```

Comment:
Please can this comment be put closer to the line it is relevant to? I spent a few minutes trying to understand line 48 only to see this afterwards 😅